pyiceberg/table/update/snapshot.py

```diff
@@ -118,6 +119,7 @@ def __init__(
         commit_uuid: Optional[uuid.UUID] = None,
         snapshot_properties: Dict[str, str] = EMPTY_DICT,
         branch: str = MAIN_BRANCH,
+        stage_only: bool = False,
```

Comment:
I think API wise, this makes more sense. In the case `branch is None` then we don't set the ref, if it is `not None` then we set the ref.
```suggestion
        branch: Optional[str] = MAIN_BRANCH
```

@kevinjqliu WDYT?

Reply:
Make sense, thanks for taking a look!